### ISSUE TITLE
Do not activate p2p if there is less than X viewers

### ISSUE THREAD

**Issue by DavidLibeau** [Chocobozzz/PeerTube#5493]
### Describe the problem to be solved

PeerTube could leak the IP address of viewer. Even if this personal data is not very sensitive, we can improve the software to minimise privacy risks.

### Describe the solution you would like

In order to improve privacy of PeerTube, a trick would be to enable p2p only if the concurrency viewer count reach a certain amount (number set in the options). For instance, you could only activate p2p if you have more than 10 peers. Furthermore a web server could easily handle this amount of viewers without the needs of p2p.

**Comment by Booteille** [comment]
> Furthermore a web server could easily handle this amount of viewers without the needs of p2p.

Only if you have no others video being watched on your server. You could have hundreds users watching a video and only a few watching another one.

**Comment by DavidLibeau** [comment]
If you have less than 10 peers on a video, there is only a few chances that your instance would be impacted by this feature as we could assume that people may not naturally be dispatched on all your videos. And even without this feature, we would already have the issue. In addition, I recall that this feature would be an option for admins.

**Comment by hollownights** [comment]
Well, if you take the "peer" out of "PeerTube" you are left with a really sad kind of tube. 

If something like that was to implemented, it should probably be up to the user to decide, much like a GDPR-tracking consent setting. _In fact_, if joining a swarm was to be treated as a tracking problem, then it should _definitely_ be treated in the light of GDPR.

**Comment by DavidLibeau** [comment]
Viewers can already opt-out from peer-to-peer. Click on "Settings" in the bottom left of PeerTube, then uncheck "Help share videos being played".

**Comment by hollownights** [comment]
Then, if joining a swarm is a tracking issue, the only thing left is to let users know about that right when they enter an instance's website, so they can decide about it upfront, after all, a hidden setting with tracking as default isn't GDPR compliant.

**Comment by DavidLibeau** [comment]
Users are already informed with a message. Loading a website have to transit the IP adresse to the web server. In all case, admins have legitimate interest of processing this data. This issue is not a GDPR issue, it is a proposal for mitigating privacy risks.

**Comment by hollownights** [comment]
> Loading a website have to transit the IP address to the web server.

Yes, and that isn't even the point of your own issue. The issue is "Having your IP in the swarm and exposed to third-parties: cool or not cool?"

> This issue is not a GDPR issue, it is a proposal for mitigating privacy risks.

I'm glad you clarified that an issue about privacy doesn't concern a legislation about privacy.

Again, this is the real question: "Having your IP in the swarm and exposed to third-parties: cool or not cool?". If the answer is '"not cool", just show a really big setting screen as soon as the user enter the page and let them decide if they wanna be exposed to third-parties (other users in the swarm) or not. Also again, hidden settings with tracking as default aren't GDPR compliant.

**Comment by DavidLibeau** [comment]
The issue is only about mitigating privacy risks.

Please note that GDPR do not cite the word "privacy" any single time and only cite "the respect for private and family life" in recital 4. This regulation also promote "free movement of data" in their title. Please do not bring any regulation in a feature proposal that do not cite any law.

**Comment by hollownights** [comment]
Ok, got it, you're trying to get away from a law by not talking about a law. 

> Please note that GDPR do not cite the word "privacy" any single time and only cite "the respect for private and family life" in recital 4.

Do not ever get in front of a judge.

**Comment by DavidLibeau** [comment]
Unfortunately I don't take advices from an account created 4 days ago… 😕
